### PR TITLE
Major network rework 3

### PR DESCRIPTION
**Pull request by ddominet**
Preparation for vxlan, zerotier, gre-tap common network fabric,
Major Fixes related zerotier.py module,
docker-compose modified,
introducing separated network infra for services

### CODE REVIEW

**Review by PiotrWodecki — CHANGES_REQUESTED** [review]
Please don't commit the .idea folder

**Review by ddominet — COMMENTED** [review]
Done

File: core/zerotier.py [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+            self.configure_default_network()
         else:
             self.prod_network = self.local_networks()[0]
-        self.device_count = None
+
+        self.reset_local_zerotier_interface()
 
     def request_local(self, path, data=None):
-        if data is None:
+        if data == None:
```
Comment (PiotrWodecki) [inline core/zerotier.py:27]: use `is` when comparing to `None`

File: requirements.txt [hunk 2]
```diff
@@ -5,4 +5,5 @@ crispy-bootstrap5==0.6
 requests==2.28.1
 django-polymorphic==3.1.0
 django-crispy-forms==1.14.0
-black==22.6.0
\ No newline at end of file
+black==22.6.0
+docker
```
Comment (PiotrWodecki) [inline requirements.txt:9]: Please specify the newest version so dependabot can help us keep the image up-to-date

File: docker-compose.yml [hunk 3]
```diff
@@ -26,17 +28,21 @@ services:
...
       - SERVER_HOSTNAME=localhost
-      
+    networks:
+      - service-network
   web:
     build: .
     command: python manage.py runserver 0.0.0.0:80
     volumes:
       - .:/code
+      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (PiotrWodecki) [inline docker-compose.yml:45]: This is incredibly dangerous as it gives **root access on the host machine** to anyone that can access this socket inside the container. This cannot go through and we need to find a workaround. More reading material on this topic https://integratedcode.us/2016/04/08/user-namespaces-sharing-the-docker-unix-socket/

**Review by PiotrWodecki — CHANGES_REQUESTED** [review]

**Comment by ddominet** [comment]
Branch Discontinued,
Fixes introduced in API branch

**Pull request closed without merging**